src/de/dkfz/b080/co/knowledge/metadata/COMetadataAccessor.groovy

```diff
@@ -65,95 +62,23 @@ class COMetadataAccessor {
-
-        // Get list of all known samples. Sort and revert them, so we can use them properly.
-        // Get rid of empty samples! Sort and reverse for first match searches. (e.g. control_abc comes before control!)
-        def listOfAll = (cfg.possibleControlSampleNamePrefixes + cfg.possibleTumorSampleNamePrefixes)
-                .findAll().sort().reverse()
-
-        String filename = file.name
-        if (useLowerCaseFilenamesForSampleExtraction)
-            filename = filename.toLowerCase()
-
-        String searchString = matchExactSampleNames ? "_" : ""
-
-        // FIRST match!
-        String sampleName = listOfAll.find { String sampleID ->
-            String _searchString = searchString
-            if (sampleID.endsWith("_")) {
-                logger.always("A sample name was given with an underscore at the end. We assume this is intentional and set matchExactSampleNames for '$sampleID'")
-                _searchString = ""
-            }
-            String[] sampleSplit = sampleID.split(StringConstants.SPLIT_UNDERSCORE)
-            if (sampleSplit.last().isInteger() && !matchExactSampleNames) {
-                logger.always("A sample name was given with an index at the end. We assume this is intentional and set allowSampleTerminationWithIndex for '$sampleID'")
-                sampleID = sampleID[0..-2]
-            }
-
-            filename.startsWith(sampleID + _searchString)
-        }
-
-        if (!matchExactSampleNames && sampleName) {
-            String[] sampleSplit = sampleName.split(StringConstants.SPLIT_UNDERSCORE)
-            if (sampleSplit.last().isInteger()) {
-                sampleName = sampleName[0..-3]
-                allowSampleTerminationWithIndex = true
-            }
-        }
-
-        String[] splitFilename = filename.split(StringConstants.SPLIT_UNDERSCORE)
-        if (sampleName == null && matchExactSampleNames)
-            return null
-        else if (sampleName == null) // Fallback, in case we do not want an exact match but did not find anything.
-            sampleName = splitFilename[0]
-        else {
-            if (sampleName.endsWith("_"))
-                sampleName = sampleName[0..-2] // In this special case, we remove the "_" char at the end.
-
-            // A sample can contain underscore characters "_". Here, we make sure, that the whole samplename is returned
-            // by including the count of underscore characters in the sample name.
-            sampleName = splitFilename[0..(sampleName.count("_"))].join("_")
-        }
-
-        if (allowSampleTerminationWithIndex) {
-            String indexPart = filename[(sampleName.size() + 1)..-1].split(StringConstants.SPLIT_UNDERSCORE)[0]
-            if (RoddyConversionHelperMethods.isInteger(indexPart)) {
-                sampleName += "_$indexPart"
-            }
-        }
-        return sampleName
-    }
 
     // TODO: Regex
     String extractSampleNameFromBamBasename(File file, ExecutionContext context) {
```

Comment:
This method is specific for the strategy. Why not just `getSampleName()` and then let the specific strategy (accessor) do the specific work and get the value from whereever?